### PR TITLE
Made Gallery.CredentialExpiration job store cursor in Azure.

### PR DESCRIPTION
**Pull request by agr**
Code for the [#3572](https://github.com/NuGet/NuGetGallery/issues/3572).

### CODE REVIEW

File: src/Gallery.CredentialExpiration/Gallery.CredentialExpiration.csproj [hunk 1]
```diff
@@ -45,12 +60,18 @@
...
+      <HintPath>..\..\packages\WindowsAzure.Storage.7.1.2\lib\net40\Microsoft.WindowsAzure.Storage.dll</HintPath>
+    </Reference>
     <Reference Include="Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Newtonsoft.Json.9.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
     </Reference>
     <Reference Include="NuGet.Services.Logging, Version=2.1.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\packages\NuGet.Services.Logging.2.1.1\lib\net452\NuGet.Services.Logging.dll</HintPath>
     </Reference>
+    <Reference Include="NuGet.Services.Storage, Version=0.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\NuGet.Services.Storage.2.1.1-dev-agr-stg-5247\lib\net452\NuGet.Services.Storage.dll</HintPath>
```
Comment (agr) [inline src/Gallery.CredentialExpiration/Gallery.CredentialExpiration.csproj]: Need to verify that it works in dev and publish stable version of the package

File: src/Gallery.CredentialExpiration/Job.cs [hunk 2]
```diff
@@ -70,6 +74,13 @@ public override bool Init(IDictionary<string, string> jobArgsDictionary)
 
                 _allowEmailResendAfterDays = JobConfigurationManager.TryGetIntArgument(jobArgsDictionary, MyJobArgumentNames.AllowEmailResendAfterDays)
                     ?? _allowEmailResendAfterDays;
+
+                string storageConnectionString = JobConfigurationManager.GetArgument(jobArgsDictionary, JobArgumentNames.DataStorageAccount);
+                string storageContainerName = JobConfigurationManager.GetArgument(jobArgsDictionary, JobArgumentNames.ContainerName);
+
+                CloudStorageAccount csa = CloudStorageAccount.Parse(storageConnectionString);
```
Comment (skofman1) [inline src/Gallery.CredentialExpiration/Job.cs]: var

**Comment by skofman1** [comment]
:shipit:

File: src/Gallery.CredentialExpiration/Gallery.CredentialExpiration.csproj [hunk 3]
```diff
@@ -45,12 +60,18 @@
...
+    <Reference Include="Microsoft.WindowsAzure.Storage, Version=7.1.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\WindowsAzure.Storage.7.1.2\lib\net40\Microsoft.WindowsAzure.Storage.dll</HintPath>
+    </Reference>
     <Reference Include="Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Newtonsoft.Json.9.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
     </Reference>
     <Reference Include="NuGet.Services.Logging, Version=2.1.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\packages\NuGet.Services.Logging.2.1.1\lib\net452\NuGet.Services.Logging.dll</HintPath>
     </Reference>
+    <Reference Include="NuGet.Services.Storage, Version=0.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (joelverhagen) [inline src/Gallery.CredentialExpiration/Gallery.CredentialExpiration.csproj]: 0.0.0.0 seems wrong

File: src/Gallery.CredentialExpiration/Job.cs (hunk 2 above)
Comment (agr) [inline src/Gallery.CredentialExpiration/Job.cs]: noooooo!

File: src/Gallery.CredentialExpiration/Gallery.CredentialExpiration.csproj (hunk 3 above)
Comment (agr) [inline src/Gallery.CredentialExpiration/Gallery.CredentialExpiration.csproj]: Done

**Pull request merged**